connect/api/src/main/java/org/apache/kafka/connect/rest/extension/ConnectClusterState.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest.extension;
+
+import org.apache.kafka.connect.rest.extension.entities.ConnectorStateDetail;
+
+import java.util.Collection;
+
+public interface ConnectClusterState {
+
+    /**
+     * Get a list of connectors currently running in this cluster. This is a full list of connectors
+     * in the cluster gathered from the current configuration.
+     */
+    Collection<String> connectors();
+
+    /**
+     * Lookup the current status of a connector.
+     *
+     * @param connName name of the connector
+     */
+    ConnectorStateDetail connectorStatus(String connName);
```

Comment:
How about just `ConnectorState`?

Reply:
ConnectorStateDetails provides both the ConnectorState and Connector TaskState. The original entity class has a name of ConnectorStateInfo but didn't want to use the same name because i thought it might be a little confusing

Reply:
Or maybe just `ConnectorDetail`?

Reply:
How about `ConnectorHealth`, since this is likely to be used by health check extensions? The `Detail` in `ConnectorStateDetail` just seems superfluous to me.

Reply:
yeah, I agree with Randall here. `Detail` is not a good fit here. Maybe you can use `ConnectorsContext`?